test/functional/eval/system_spec.lua

```diff
@@ -446,8 +446,11 @@ describe('systemlist()', function()
     after_each(delete_file(fname))
 
     it('replaces NULs by newline characters', function()
-      if helpers.pending_win32(pending) then return end
-      eq({'part1\npart2\npart3'}, eval('systemlist("cat '..fname..'")'))
+      if iswin() then
+        eq({'part1\npart2\npart3\r'}, eval('systemlist("type '..fname..'")'))
```

Comment:
Is the last carriage return intentional for `systemlist` to be compatible with Vim?

Reply:
`systemlist()` splits the output from the command on `\n`.  Since you're on Windows, nvim is defaulting `'fileformat'` to `dos`.  This means that the file created by `create_file_with_nuls()` has a `\r\n` for the line ending instead of the `\n` that's expected on non-Windows systems.

Therefore, what you're seeing is expected given the differences in default `'fileformat'`.  You can either account for it here, like you did, or change the test setup to force `fileformat=unix`.